### PR TITLE
Best practices for k8s

### PR DESCRIPTION
**Pull request by Arau**
- DaemonSet no Helm
- external etcd
- Mixed compute only with full installation based on labels

The aim of the PR is to get feedback and review the doc rather than merging to the docs.

### CODE REVIEW

File: assets/production-bestpractices.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+
+# StorageOS production installation
+
+*Installation with DaemonSets on Kubernetes*
+
+StorageOS tries to keep the installation process simple for users, requiring only minimal configuration. However, more options are available if you are deploying in complex environments, or where tuning for availability and performance is desired.  This guide discusses deployment options to consider when running in production.
+
+Checkout the StorageOS Helm [chart](https://github.com/storageos/helm-chart) for an easy automated installation.
```
Comment (oicheryl) [inline assets/production-bestpractices.md]: Checkout => Check out
Link the full "StorageOS Helm Chart", not just "chart".

File: assets/production-bestpractices.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+StorageOS ships an embedded key value store service based on etcd. The kv store is used to keep cluster consistency among the StorageOS nodes. By default, the first 5 nodes of StorageOS will deploy the etcd servers. Additional nodes don't start an etcd server, only the client component. Since DaemonSets start pods without considering the order, it is not possible to define which cluster nodes will start the etcd servers. Since the failure of the majority of the etcd servers (half of the etcd servers +1) could cause a lost in quorum, the StorageOS cluster could become unstable until the nodes are restored. It is recommended to keep the etcd servers as stable as possible. To guarantee high availability it is recommended to use one of the two following options.
+
+1. Use an external etcd cluster
+2. Run a compute only StorageOS daemonset where the etcd masters will run based on node labels. 
+
+## External etcd cluster (+v3.0).
+
+Deploy an external etcd cluster in high availability. Either 3 or 5 nodes to keep quorum.
```
Comment (oicheryl) [inline assets/production-bestpractices.md]: ". Either" => ", either"

File: assets/production-bestpractices.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+kubectl label node my_node1 my_node2 my_node3 storageos=compute-only
+```
+
+Then label the rest of the nodes that will run StorageOS:
+
+```
+kubectl label node my_nodeX storageos=storage
+```
+
+> Nodes not labeled won't run StorageOS in this installation procedure. Pods running on nodes where StorageOS is not present cannot mount volumes managed by StorageOS.
```
Comment (oicheryl) [inline assets/production-bestpractices.md]: "Nodes not labeled" => "Unlabelled nodes"

File: assets/production-bestpractices.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+Checkout the StorageOS Helm [chart](https://github.com/storageos/helm-chart) for an easy automated installation.
+
+StorageOS ships an embedded key value store service based on etcd. The kv store is used to keep cluster consistency among the StorageOS nodes. By default, the first 5 nodes of StorageOS will deploy the etcd servers. Additional nodes don't start an etcd server, only the client component. Since DaemonSets start pods without considering the order, it is not possible to define which cluster nodes will start the etcd servers. Since the failure of the majority of the etcd servers (half of the etcd servers +1) could cause a lost in quorum, the StorageOS cluster could become unstable until the nodes are restored. It is recommended to keep the etcd servers as stable as possible. To guarantee high availability it is recommended to use one of the two following options.
+
+1. Use an external etcd cluster
+2. Run a compute only StorageOS daemonset where the etcd masters will run based on node labels. 
+
+## External etcd cluster (+v3.0).
```
Comment (oicheryl) [inline assets/production-bestpractices.md]: Option 1: External etcd cluster (+v3.0)

File: assets/production-bestpractices.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+```
+        env:
+           - name: KV_BACKEND
+             value: 'etcd'
+           - name: KV_ADDR
+             value: 'etcd-storageos:2379'
+```
+
+## Compute Only DaemonSet
```
Comment (oicheryl) [inline assets/production-bestpractices.md]: Option 2: Compute Only DaemonSet

File: assets/production-bestpractices.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+Checkout the StorageOS Helm [chart](https://github.com/storageos/helm-chart) for an easy automated installation.
+
+StorageOS ships an embedded key value store service based on etcd. The kv store is used to keep cluster consistency among the StorageOS nodes. By default, the first 5 nodes of StorageOS will deploy the etcd servers. Additional nodes don't start an etcd server, only the client component. Since DaemonSets start pods without considering the order, it is not possible to define which cluster nodes will start the etcd servers. Since the failure of the majority of the etcd servers (half of the etcd servers +1) could cause a lost in quorum, the StorageOS cluster could become unstable until the nodes are restored. It is recommended to keep the etcd servers as stable as possible. To guarantee high availability it is recommended to use one of the two following options.
+
+1. Use an external etcd cluster
+2. Run a compute only StorageOS daemonset where the etcd masters will run based on node labels. 
+
```
Comment (oicheryl) [inline assets/production-bestpractices.md]: Explain pros and cons of each option. (One line is enough.)

File: assets/production-bestpractices.md (hunk 1 above)
Comment (Arau) [reply assets/production-bestpractices.md]: done

File: assets/production-bestpractices.md (hunk 2 above)
Comment (Arau) [reply assets/production-bestpractices.md]: done

File: assets/production-bestpractices.md (hunk 3 above)
Comment (Arau) [reply assets/production-bestpractices.md]: done

File: assets/production-bestpractices.md (hunk 4 above)
Comment (Arau) [reply assets/production-bestpractices.md]: done

File: assets/production-bestpractices.md (hunk 5 above)
Comment (Arau) [reply assets/production-bestpractices.md]: done

**Comment by actionbuddha** [comment]
The content here looks great but given our new structure I'm unconvinced that a single 'best practices' section makes sense across all architectures. I wonder if this lends itself to a similar approach to troubleshooting, where a series of includes shows under each platform as appropriate...

**Pull request closed without merging**